### PR TITLE
Refactored to use Cassandra 3.9, Spark 2.1.0, Presto .167 and Yanagis…

### PR DESCRIPTION
**Pull request by BrentDorsey**
Refactored to use Cassandra 3.9, Spark 2.1.0, Presto .167 and Yanagishima Presto UI. 

Implemented base Maven POM for existing modules.

education.ml Dockerfile updates to latest version:
•	CASSANDRA_VERSION=3.9
•	SPARK_OTHER_VERSION=2.1.0
•	PRESTO_VERSION=0.167
•	SPARK_CASSANDRA_CONNECTOR_VERSION=1.6.5
•	SPARK_AVRO_CONNECTOR_VERSION=3.2.0
•	SBT_SPARK_PACKAGES_PLUGIN_VERSION=0.2.5
•	SPARK_XML_VERSION=0.3.5

Implemented two new generic secrets in every ReplicationController yaml that mounts the git-volume:
•	github ssh key secret to enable the git-volume mount bound to Enterprise github repo
•	kube-config-secret-volume to eliminate the need to bash into pods and manually populate .kube/config

$ kubectl create secret generic githubsshkey \
  --from-file=private-key=/root/.ssh/id_rsa \
  --from-file=publickey=/root/.ssh/id_rsa.pub

ReplicationController.yaml
imagePullSecrets:
  - name: githubsshkey

$ kubectl create secret generic kubeconfig --from-file=.kube/config

ReplicationController.yaml
  volumeMounts:
    - name: kube-config-secret-volume
      readOnly: true
      mountPath: /root/.kube
volumes:
  - name: kube-config-secret-volume
    secret:
      secretName: kube-config-secret

### CODE REVIEW

File: dashboard.ml/hystrix/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+hystrix-dashboard
+=======================================================
+
+# TODO FIX dashboard.ml/hystrix README
+
```
Comment (cfregly) [inline dashboard.ml/hystrix/README.md]: did this slip in somehow?

File: dashboard.ml/hystrix/pom.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (cfregly) [inline dashboard.ml/hystrix/pom.xml]: and this?

File: dashboard.ml/kubernetes-ec2-autoscaler/pom.xml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>pipeline.io</groupId>
```
Comment (cfregly) [inline dashboard.ml/kubernetes-ec2-autoscaler/pom.xml]: how about this?

File: package.ml/presto/0.161/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,35 @@
+FROM fluxcapacitor/package-java-openjdk-1.8
+
+WORKDIR /root
+
+ENV \
```
Comment (cfregly) [inline package.ml/presto/0.161/Dockerfile]: is presto 1.6.1 needed?

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?><!-- This project was generated from HomeAway's war-basic-archetype. --><project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+
+	<groupId>pipeline.io</groupId>
```
Comment (cfregly) [inline pom.xml]: is this needed?  root-level pom.xml?

File: presto.ml/yanagishima-svc.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: yanagishima
```
Comment (cfregly) [inline presto.ml/yanagishima-svc.yaml]: i'm gonna rename this to just presto-ui or something

File: presto.ml/yanagishima/conf/yanagishima.properties [hunk 7]
```diff
@@ -0,0 +1,8 @@
+jetty.port=4125
+presto.coordinator.server=http://presto-master:7060
+presto.redirect.server=http://presto-master:7060
+select.limit=500
+catalog=cassandra
```
Comment (cfregly) [inline presto.ml/yanagishima/conf/yanagishima.properties]: i assume the cassandra config is elsewhere?

**Comment by BrentDorsey** [comment]
I added a readme for hystrix, the TODO was to add more detail.

**Comment by BrentDorsey** [comment]
No you can remove it.  Was experimenting to determine the latest version that works with airpal which of course is 0.145

**Comment by BrentDorsey** [comment]
Not required, you must not like maven eh?

Feel free to remove it.

**Comment by BrentDorsey** [comment]
Go for it, yanagishima is freaking hard to spell.

**Comment by BrentDorsey** [comment]
I thought you were supposed to be attending Strata ☺.

You must be bored and already know everything they’re discussing ahahahahahaha

**Comment by BrentDorsey** [comment]
Katsumi Yanagishima<https://en.wikipedia.org/wiki/Katsumi_Yanagishima> – Japanese cinematographer (that codes hahaha j/k)
Yanagishima<https://www.google.com/maps/search/yanagishima/@35.0795363,138.1163855,9z> – google maps

[cid:image001.png@01D29DA4.D2EE36B0]

From: Brent Dorsey <bdorsey@homeaway.com>
Date: Wednesday, March 15, 2017 at 3:50 PM
To: fluxcapacitor/pipeline <reply@reply.github.com>
Subject: Re: [fluxcapacitor/pipeline] Refactored to use Cassandra 3.9, Spark 2.1.0, Presto .167 and Yanagis… (#208)

FYI – Yanagishima is display in the UI as well, he/she obviously liked using their name or that work as it’s everywhere.

**Pull request merged**